src/rgw/rgw_rest_log.cc

```diff
@@ -78,10 +78,15 @@ void RGWOp_MDLog_List::execute() {
   } 
 
   if (period.empty()) {
-    ldout(s->cct, 5) << "Missing period id" << dendl;
-    http_ret = -EINVAL;
-    return;
+    ldout(s->cct, 5) << "Missing period id trying to use current" << dendl;
+    period = store->get_current_period_id();
+    if (period.empty()) {
+      ldout(s->cct, 5) << "Missing period id" << dendl;
+      http_ret = -EINVAL;
+      return;
+    }
```

Comment:
@cbodley this is for the sync agent, which doesn't send a period id at all. The idea is to allow a path for migration from old config to new one without breaking the sync agent. That means that if users created a new configuration with a realm and a period, then we should still allow it to continue working as it was working before.